contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol

```diff
@@ -315,23 +303,25 @@ abstract contract FunctionsBilling is HasRouter, IFunctionsBilling {
 
     bytes32 requestId = computeRequestId(address(this), billing.client, billing.subscriptionId, initiatedRequests + 1);
 
-    Commitment memory commitment = Commitment(
-      billing.subscriptionId,
+    IFunctionsRequest.Commitment memory commitment = IFunctionsRequest.Commitment(
+      adminFee,
+      address(this),
       billing.client,
+      billing.subscriptionId,
       billing.callbackGasLimit,
-      address(this),
-      adminFee,
       estimatedCost,
+      uint40(block.timestamp + s_config.requestTimeoutSeconds),
+      requestId,
       donFee,
-      uint32(block.timestamp),
-      s_config.gasOverheadBeforeCallback + s_config.gasOverheadAfterCallback,
-      billing.expectedGasPrice
+      s_config.gasOverheadBeforeCallback,
+      s_config.gasOverheadAfterCallback
     );
-    s_requestCommitments[requestId] = commitment;
 
-    emit BillingStart(requestId, commitment);
+    s_requestCommitments[requestId] = keccak256(abi.encode(commitment));
+
+    emit BillingStart(requestId, estimatedCost);
```

Comment:
Why do we emit only the estimated cost?  Would it be more helpful to emit the entire commitment?
Does it make sense to just omit this event entirely given all the data will also be contained in the OracleRequest event?

Reply:
I took it out for that exact reason, it's already on OracleRequest.
I'm actually open to removing the billing events, we don't actually consume them anywhere.

Reply:
Reverting this actually, to be handled on another PR

Reply:
Yeah, if we can remove them I think we should. Less code = less problems, right?